### ISSUE TITLE
One-off no-content templates in content directory

### ISSUE THREAD

**Issue by ge0ffrey** [jbake-org/jbake#693]
I have a team.html page which list all our contributors.
That needs to get generated through an `ftl` template that reads from `data/users.yml`. There is no other html page like it, that would reuse this layout. It's a _one-off_ template.

I have several other pages (download.html, documentation.html) that require a one-off template.
**In fact, index.html is a natural one-off template.** So is feed, archive, etc. Post is not.

**What I need to do now:**

1) Create a file `content/community/team.html`:

```
title=Team
type=community_team
~~~~~~
```

2) In `jbake.properties` add:

```
template.community_team.file=community_team.ftl
```

3) Create a file `templates/community_team.ftl`:

```
<#import "base.ftl" as base>
<@base.layout>
    <h1>Team</h1>
    <#assign users = data.get('users.yml').data>
    <#list users as user>
       <p>${user.name}</p>
    </#list>
</@base.layout>
```

Proposal A) **What I want to do:**

1) Create a file `content/community/team.html.ftl`:

```
<#import "base.ftl" as base>
<@base.layout title="Team">
    <h1>Team</h1>
    <#assign users = data.get('users.yml').data>
    <#list users as user>
       <p>${user.name}</p>
    </#list>
</@base.layout>
```

**Comment by ge0ffrey** [comment]
Update: `tags` must remain as a build-in template, because it creates one per page per tag

**Comment by jonbullock** [comment]
Gotcha, need to work out how this fits into the pipeline (does this file get added to the content store as a piece of "content", does it actually need the metadata header etc.). Should have a better idea of the scope of work required then.

**Comment by LightGuard** [comment]
It should be able to access the same information as a template. Is it a piece of content? I would say yes.

**Comment by ge0ffrey** [comment]
@jonbullock Any progress on this issue? It's the number one issue for me.

**Comment by jonbullock** [comment]
@ge0ffrey No progress yet, all I can say is that it's still on my to do list.

**Comment by tarilabs** [comment]
Hi, if this helps I've experimented with an alternative approach which _works_ (with limitation of file-extension, below) but does not require the workaround to define a single template instance file per each one-offs.

You can see it in action here:

- commit [7713a9549802167ba9d5db37b3ca109976fa1f7c](https://github.com/tarilabs/matteomortari/pull/1/commits/7713a9549802167ba9d5db37b3ca109976fa1f7c) using workaround as detailed in this thread

- commit [4365065a7e43923702133129b42b4f6bfe7c20d6](https://github.com/tarilabs/matteomortari/pull/1/commits/4365065a7e43923702133129b42b4f6bfe7c20d6) avoid the need of "a template file instance for each one-off" by using a generic template with FreeMarker `interpret` (see [interpret doc here)](https://freemarker.apache.org/docs/ref_builtins_expert.html#ref_builtin_interpret)

The downside: in `/content` the file is `.html` and not `.ftl` as the file extensions which can be baked from content seems to me is fixed (and due to constraint as [detailed here](https://jbake.org/docs/2.6.7/#static_files_in_content:~:text=Any%20files%20other%20than%20above%20listed%20content%20files%2C%20will%20be%20treated%20as%20static%20files))

In other words I edit the `/content/oneoff.html` as if it's a `.ftl` and then I wire it into a **_single_** FTL template designed on purpose, which interprets the content body as if it was part of the originating FTL.
